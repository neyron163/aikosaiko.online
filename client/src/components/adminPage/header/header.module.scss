@import '../../styleguide//variables.scss';

:local {
    .container {
        height: 48px;
        background: var(--color-bay-of-many);
    }

    .logo {
        font-size: 20px;
        color: var(--color-white);
        font-weight: 700;
    }

    .button {
        color: var(--color-white); 
    }

    .flex {
        display: flex;
        max-width: 1440px;
        margin: 0 auto;
        height: 100%;
    }

    .leftSide, .rightSide {
        width: 50%;
        display: flex;
        justify-content: flex-end;
        align-items: center;
    }
}