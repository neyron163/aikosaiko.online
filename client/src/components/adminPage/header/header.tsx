import React from 'react';
import SvgIcon from '@material-ui/core/SvgIcon';
import {LinkButton} from 'components/linkButton/linkButton';

import s from './header.module.scss';

export const Header = () => (
    <header className={s.container}>
        <div className={s.flex}>
            <div className={s.leftSide}>
                <div className={s.logo}>Admin Page</div>
            </div>
            <div className={s.rightSide}>
                <LinkButton path="/" icon="fas fa-home" className={s.button}>
                    Home page
                </LinkButton>
            </div>
        </div>
    </header>
);
