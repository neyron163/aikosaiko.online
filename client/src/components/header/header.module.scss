@import '../styleguide/variables.scss';

:local {
    .header {
        height: 250px;
        overflow: hidden;
        display: flex;
        flex-direction: column;
        justify-content: flex-end;
        position: relative;
    }

    .img {
        position: absolute;
        min-width: 105%;
        left: 50%;
        top: 0;
        transform: translateX(-50%);
    }

    .menu {
        position: relative;
        background: var(--color-mine-shaft);
        height: 50px;
    }

    .list {
        display: flex;

        li {
            margin-right: 10px;

            &:last-child {
                margin-right: 0;
            }
        }

        a {
            color: var(--color-white);
        }
    }

    .grid {
        max-width: 1220px;
        margin: 0 auto;
        display: flex;
        align-items: center;
        height: 100%;
    }
}
