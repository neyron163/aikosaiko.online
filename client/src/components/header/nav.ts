export const menu = [
    {
        text: 'Главная',
        path: '/',
    },
    {
        text: 'Новости',
        path: '/news',
    },
    {
        text: 'Пожертвование',
        path: '/charity',
    },
    {
        text: 'admin',
        path: '/admin',
    },
];
