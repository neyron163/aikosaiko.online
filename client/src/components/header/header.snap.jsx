import React from 'react';
import renderer from 'react-test-renderer';

import {Header} from './header';

jest.mock('../linkButton/linkButton', () => ({LinkButton: 'LinkButton'}));

describe('<Header />', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly if night true', () => {
        const tree = renderer.create(<Header night />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
