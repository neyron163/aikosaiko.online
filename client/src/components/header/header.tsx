import React from 'react';

import white from './images/white.jpg';
import dark from './images/dark.jpg';
import {LinkButton} from '../linkButton/linkButton';
import {nav} from './nav';

import s from './header.module.scss';

interface PropsType {
    night?: boolean;
}
interface NavType {
    text: string;
    path: string;
}

/**
 * Header component
 * @param {Header} - props
 */
export const Header: React.FC<PropsType> = ({night = false}) => (
    <header className={s.header}>
        <img className={s.img} src={night ? dark : white} alt="header" />
        <nav className={s.menu}>
            <div className={s.grid}>
                <ul className={s.list}>
                    {nav.map(({text, path}: NavType) => (
                        <li key={text}>
                            <LinkButton path={path}>{text}</LinkButton>
                        </li>
                    ))}
                </ul>
            </div>
        </nav>
    </header>
);
