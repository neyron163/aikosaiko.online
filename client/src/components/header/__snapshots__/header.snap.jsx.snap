// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<header
  className="header"
>
  <img
    alt="header"
    className="img"
    src="white.jpg"
  />
  <nav
    className="menu"
  >
    <div
      className="grid"
    >
      <ul
        className="list"
      >
        <li>
          <LinkButton
            path="/"
          >
            Главная
          </LinkButton>
        </li>
        <li>
          <LinkButton
            path="/news"
          >
            Новости
          </LinkButton>
        </li>
        <li>
          <LinkButton
            path="/charity"
          >
            Пожертвование
          </LinkButton>
        </li>
        <li>
          <LinkButton
            path="/admin"
          >
            admin
          </LinkButton>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;

exports[`<Header /> should render correctly if night true 1`] = `
<header
  className="header"
>
  <img
    alt="header"
    className="img"
    src="dark.jpg"
  />
  <nav
    className="menu"
  >
    <div
      className="grid"
    >
      <ul
        className="list"
      >
        <li>
          <LinkButton
            path="/"
          >
            Главная
          </LinkButton>
        </li>
        <li>
          <LinkButton
            path="/news"
          >
            Новости
          </LinkButton>
        </li>
        <li>
          <LinkButton
            path="/charity"
          >
            Пожертвование
          </LinkButton>
        </li>
        <li>
          <LinkButton
            path="/admin"
          >
            admin
          </LinkButton>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;

exports[`<Header /> should render correctly if pathname = admin 1`] = `null`;
